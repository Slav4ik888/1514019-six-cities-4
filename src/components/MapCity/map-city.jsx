import React from 'react';
import pt from 'prop-types';

const MapCity = ({children}) => {
  return (
    <>
    {children}
    </>
  );
};

MapCity.propTypes = {
  children: pt.node.isRequired,
};

export default MapCity;
