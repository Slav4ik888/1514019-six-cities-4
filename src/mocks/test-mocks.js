export const testOffers = [
  {
    id: 1,
    isPremium: true,
    isFavourite: true,
    previewImage: `img/apartment-01.jpg`,
    price: 55,
    rating: 4.8,
    offerType: `Apartment`,
    cardTitle: `Test name-1`,
    coordinates: [52.3509553943508, 4.97309666406198],

  },
  {
    id: 2,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-02.jpg`,
    price: 250,
    rating: 2,
    offerType: `Hotel`,
    cardTitle: `Test name-2`,
    coordinates: [52.3609553943508, 4.95309666406198],

  }
];

export const testOffer = {
  id: 1,
  isPremium: true,
  isFavourite: true,
  previewImage: `img/apartment-01.jpg`,
  price: 55,
  rating: 4.8,
  offerType: `Private room`,
  cardTitle: `Test name-1`,
  coordinates: [52.3709553943508, 4.78309666406198],

};
