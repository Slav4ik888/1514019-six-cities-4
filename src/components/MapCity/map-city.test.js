import React from 'react';
import renderer from 'react-test-renderer';
import MapCity from './map-city.jsx';
import {testOffers} from '../../mocks/test-mocks.js';


it(`check render MapCity`, () => {
  const component = renderer.create(
      <MapCity
        offers={testOffers}
        activeCoords={[51.200569, 6.787428]}
      />, {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }
  ).toJSON();

  expect(component).toMatchSnapshot();
});
