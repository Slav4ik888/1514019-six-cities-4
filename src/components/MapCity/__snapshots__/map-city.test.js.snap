// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render MapCity 1`] = `<div />`;
